yt_dlp/extractor/manyvids.py

```diff
@@ -1,161 +1,64 @@
-import re
+import json
 
 from .common import InfoExtractor
-from ..utils import (
-    determine_ext,
-    extract_attributes,
-    int_or_none,
-    str_to_int,
-    url_or_none,
-    urlencode_postdata,
-)
 
 
 class ManyVidsIE(InfoExtractor):
     _VALID_URL = r'(?i)https?://(?:www\.)?manyvids\.com/video/(?P<id>\d+)'
     _TESTS = [{
         # preview video
-        'url': 'https://www.manyvids.com/Video/133957/everthing-about-me/',
-        'md5': '03f11bb21c52dd12a05be21a5c7dcc97',
+        'url': 'https://www.manyvids.com/Video/1856601/Horny-Asian-Bunny-Needs-Cock/',
         'info_dict': {
-            'id': '133957',
+            'id': '1856601',
             'ext': 'mp4',
-            'title': 'everthing about me (Preview)',
-            'uploader': 'ellyxxix',
-            'view_count': int,
+            'title': 'Horny Asian Bunny Needs Cock',
+            'description': "This naughty little Asian bunny is bored and horny. Feed her something hard and big.",
+            'uploader': 'nicoledoshi',
             'like_count': int,
         },
     }, {
         # full video
-        'url': 'https://www.manyvids.com/Video/935718/MY-FACE-REVEAL/',
-        'md5': 'bb47bab0e0802c2a60c24ef079dfe60f',
+        'url': 'https://www.manyvids.com/Video/4957054/Latex-Dress-JOI/',
         'info_dict': {
-            'id': '935718',
+            'id': '4957054',
             'ext': 'mp4',
-            'title': 'MY FACE REVEAL',
-            'description': 'md5:ec5901d41808b3746fed90face161612',
-            'uploader': 'Sarah Calanthe',
-            'view_count': int,
+            'title': 'Latex Dress JOI',
+            'description': "My new latex dress feels so good on my skin! Just wearing it for you makes me horny. Jerk your cock while I show off my latex-clad body from all angles, rubbing my hands over my body and describing how good it feels. I can't help but touch my pussy, and I give you a countdown when I'm close so we can cum together for my new latex dress!",
+            'uploader': "Kylee Nash",
             'like_count': int,
         },
     }]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
-        real_url = 'https://www.manyvids.com/video/%s/gtm.js' % (video_id, )
-        try:
-            webpage = self._download_webpage(real_url, video_id)
-        except Exception:
-            # probably useless fallback
-            webpage = self._download_webpage(url, video_id)
+        infos_url = f'https://www.manyvids.com/bff/store/video/{video_id}'
+        info_webpage = self._download_webpage(infos_url, video_id, expected_status=200)
+        info: dict = json.loads(info_webpage).get('data')
```

Comment:
- don't use type hints in extractor code
- `expected_status=200` is unnecessary
- use `_download_json()`
- if this is just for metadata and should be non-fatal, use `traverse_obj` instead of `get`. you can expect the type and fallback to an empty dict
```suggestion
        info = traverse_obj(
            self._download_json(f'https://www.manyvids.com/bff/store/video/{video_id}', video_id),
            ('data', {dict})) or {}
```